pkg/process/process.go

```diff
@@ -0,0 +1,90 @@
+package process
+
+import (
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/pkg/errors"
+)
+
+type Manager interface {
+	Pid() (int, error)
+	Name() string
+	PidFilePath() string
+	Exists() bool
+	Kill() error
+	FindProcess() (*os.Process, error)
+	WritePidFile(pid int) error
+}
+
+type Process struct {
+	name        string
+	pidFilePath string
+}
+
+func New(name, pidFilePath string) (*Process, error) {
+	return &Process{name: name, pidFilePath: pidFilePath}, nil
+}
+
+func (p *Process) Name() string {
+	return p.name
+}
+
+func (p *Process) PidFilePath() string {
+	return p.pidFilePath
+}
+
+func (p *Process) Pid() (int, error) {
+	data, err := os.ReadFile(p.PidFilePath())
+	if err != nil {
+		return -1, err
+	}
+	pidStr := strings.TrimSpace(string(data))
+	pid, err := strconv.Atoi(pidStr)
+	return pid, errors.Wrap(err, "invalid pid file")
+}
+
+func (p *Process) FindProcess() (*os.Process, error) {
+	pid, err := p.Pid()
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot find process")
+	}
+
+	proc, err := os.FindProcess(pid)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot find process")
+	}
+	return proc, nil
+}
+
+func (p *Process) Exists() bool {
+	proc, err := p.FindProcess()
+	if err != nil {
+		return false
+	}
+	if err := proc.Signal(syscall.Signal(0)); err != nil {
```

Comment:
Why `syscall.Signal(0)`? Could this use one of the `SIGXXX` constants from https://pkg.go.dev/syscall#pkg-constants ? (search for `SIGABRT`)

Reply:
Is it safe to call SIGABRT on the running process? On searching I found sending 0 is the safest way to check a process is running

Reply:
I mentioned `SIGABRT` as a way to find the relevant section, I was not suggesting to use this signal :)
However, if `FindProcess()` found it, I’d assume it means the process is running, no need to check again that it’s running ?